web3_multi_provider/async_multi_http_provider.py

```diff
@@ -0,0 +1,129 @@
+import logging
+from abc import ABC
+from typing import Any
+
+from eth_typing import URI
+from web3 import AsyncHTTPProvider
+from web3._utils.empty import Empty, empty
+from web3.providers.async_base import AsyncJSONBaseProvider
+from web3.providers.rpc.utils import ExceptionRetryConfiguration
+from web3.types import RPCEndpoint, RPCResponse
+
+from web3_multi_provider.poa import sanitize_poa_response
+from web3_multi_provider.exceptions import NoActiveProviderError, ProtocolNotSupported
+
+logger = logging.getLogger(__name__)
+
+
+
```

Comment:
Pls, reformat file with `black .`
Also, check manually `isort`, `mypy` and `pylint` outputs.
We will also check them before publishing. Later going to add github wf for this.

Reply:
Done - there was a bunch of pre-existing pylint issues with the tests, but I guess those aren't too important. It flagged a duplicate code issue with the new async providers. As mentioned it's not super straightforward to refactor these to avoid duplication, so I have disabled that warning in that file for now.